app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt

```diff
@@ -55,4 +59,9 @@ class RepositoryModule {
     @Singleton
     fun providesTransactionRepository(gatewayApi: GatewayApi): TransactionRepository =
         TransactionRepository(gatewayApi)
+
+    @Provides
+    @Singleton
+    fun providesBip39Generator(@ApplicationContext context: Context): Bip39Generator =
```

Comment:
already provided in the ApplicationModule

Reply:
Have you tried using this dependency in a `viewModel` ? This dependency is not visible within the `ViewModelFactoryModule`.

Reply:
If this generator is used only in viewModels we should remove provide statements from application module. 
```
 @Provides
    fun providesBip39(wordListProvider: WordListProvider): Bip39 = Bip39Generator(wordListProvider)

    @Provides
    fun providesWordListProvider(@ApplicationContext context: Context): WordListProvider = AndroidWordListProvider(context)

    @Provides
    fun providesMnemonicKeyAndAddressDerivator(bip39: Bip39): MnemonicKeyAndAddressDerivator = MnemonicKeyAndAddressDerivator(bip39)
```  could be moved to RepositoryModule then